### ISSUE TITLE
compute_ic_hac_stats applies a small-sample correction by default and does not report it

### ISSUE THREAD

**Issue by stefan-jansen** [ml4t/diagnostic#79]
**Version:** ml4t-diagnostic 0.1.6.post2

`ml4t.diagnostic.metrics.ic_inference.compute_ic_hac_stats` has `use_correction=True` by default. The returned dict reports `used_naive_fallback`, `effective_lags` and the naive statistics, but not whether the correction was applied, so a result cannot be reproduced or compared with another HAC implementation from its output alone.

60 periods of simulated IC:

```
default:              hac_se 0.014522, t_stat 1.9094, p_value 0.0611
use_correction=False: hac_se 0.014401, t_stat 1.9256, p_value 0.0590
```

Here the p-value moves across 0.06, and with short samples the difference is larger. Code that ports an existing Newey-West computation (statsmodels without a correction, for example) gets different numbers with no signal why.

**Expected:** the result includes `use_correction` (and the kernel), and the docstring states which correction is applied.

**Comment by stefan-jansen** [comment]
Confirmed on 0.1.6.post2 through the public `ml4t.diagnostic.metrics.compute_ic_hac_stats` API: `use_correction=True` and `False` produce different HAC statistics, but both result dictionaries omit the correction setting and kernel. This is accepted as a backward-compatible metadata fix. The implementation will return both settings on successful, undersized-input, and explicit naive-fallback paths, and document that statsmodels applies the finite-sample covariance factor `n / (n - k)`, with `k = 1` for this intercept-only mean test.
